src/analysis/normed_space/riesz_lemma.lean

```diff
@@ -51,3 +55,33 @@ begin
     ... ≤ dist x (y₀ + y) : metric.inf_dist_le_dist_of_mem hy₀y
     ... = ∥x - y₀ - y∥ : by { rw [sub_sub, dist_eq_norm] }
 end
+
+/--
+A version of Riesz lemma: given a strict closed subspace `F`, one may find an element of norm `≤ R`
+which is at distance  at least `1` of every element of `F`. Here, `R` is any given constant
+strictly larger than the norm of an element of norm `> 1`. For a version without an `R`, see
+`riesz_lemma`. -/
+lemma riesz_lemma_of_norm_lt
+  {c : 𝕜} (hc : 1 < ∥c∥) {R : ℝ} (hR : ∥c∥ < R)
+  {F : subspace 𝕜 E} (hFc : is_closed (F : set E)) (hF : ∃ x : E, x ∉ F) :
+  ∃ x₀ : E, ∥x₀∥ ≤ R ∧ ∀ y ∈ F, 1 ≤ ∥x₀ - y∥ :=
```

Comment:
Why is `c` needed here? What happens if you just have the hypothesis `1 < R`? Is it possible that there is no `c : 𝕜` with `1 < ∥c∥ < R`?

Also, I would love to have the answer to those questions in the docstring of the lemma :)

Reply:
Yes, there can be a gap in possible norms. For instance, in Q_p, the norms of the elements are of the form p^n, so there is no element of norm strictly between 1 and p. I have expanded the docstring.

Reply:
Thanks!